### PR TITLE
LGA-2984 - Adding tell us about your problem mi report

### PR DESCRIPTION
**Pull request by said-moj**
## What does this pull request do?

Adding tell us about your problem mi report.

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [x] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

File: cla_backend/apps/reports/urls.py [hunk 1]
```diff
@@ -40,6 +40,11 @@
     url(r"^mi-eod-extract/$", views.mi_eod_extract, name="mi_eod_extract"),
     url(r"^mi-comlpaints/$", views.mi_complaints, name="mi_complaints"),
     url(r"^mi-obiee-extract/$", views.mi_obiee_extract, name="mi_obiee_extract"),
+    url(
+        r"^mi-tell-us-about-your-problem/$",
+        views.tell_us_more_about_your_problem,
+        name="mi_tell_us_about_your_problem",
```
Comment (BenMillar-CGI) [inline cla_backend/apps/reports/urls.py:46]: Is the title of this report intended to be 'Tell us more about your problem'? 
In the [ticket](https://dsdmoj.atlassian.net/jira/software/c/projects/LGA/boards/226?selectedIssue=LGA-2984) this report was given the title 'Problem categorisation'

File: cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      ELSE true
+   END as "Adjustments Language"
+   ,COALESCE(adapt.no_adaptations_required, false)::bool as "Adjustments Not required"
+  ,COALESCE(adapt.minicom, false)::bool as "Adjustments Minicom"
+  ,COALESCE(adapt.text_relay, false)::bool as "Adjustments Text Relay"
+  ,COALESCE(adapt.skype_webcam, false)::bool as "Adjustments Skype"
+  ,COALESCE(log_changed_category.diagnosis_category, category.code) as "Diagnosis Category"
+  ,category.code as "Legalaid Category Code"
+  ,category.name as "Legalaid Category Name"
+  ,category.description as "Legalaid Category Description"
```
Comment (BenMillar-CGI) [inline cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql:31]: It is helpful to export this information? It often is null and is sometimes a paragraph describing the legal category.

File: cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    FROM cla_eventlog_log as log_created
+    JOIN cla_eventlog_log as log_changed_category ON log_created.case_id=log_changed_category.case_id
+        AND log_created.notes = 'Case created digitally' AND log_changed_category.notes LIKE 'Changed category to%%'
+)
+SELECT
+  c.personal_details_id as "Personal Details Id"
+  ,c.reference as "Case Id"
+  ,c.source as "Source"
+  ,to_char(c.created, 'YYYY-MM-DD') as "Created"
+  ,to_char(c.modified, 'YYYY-MM-DD') as "Modified"
```
Comment (BenMillar-CGI) [inline cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql:12]: Should these above three fields be exported, as they are not part of the report spec.

**Review by BenMillar-CGI — APPROVED** [review]
Looks good to me 👍 
I have left some queries, however, feel free to dismiss them if you disagree as they do not impact the report functionality.

**Review by TawneeOwl — APPROVED** [review]
LGTM

File: cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql (hunk 2 above)
Comment (TawneeOwl) [reply cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql:31]: Removed category description

File: cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql (hunk 3 above)
Comment (TawneeOwl) [reply cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql:12]: Removed source as not part of spec; have kept in created and modified as date information is always useful in reports

**Review by BenMillar-CGI — DISMISSED** [review]
LGTM, thank you 👍

File: cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql (hunk 3 above)
Comment (said-moj) [reply cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql:11]: @TawneeOwl @BenMillar-CGI the source field was requested by Rachel after demoing the work to her

File: cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql [hunk 4]
```diff
@@ -28,7 +27,6 @@ SELECT
   ,COALESCE(log_changed_category.diagnosis_category, category.code) as "Diagnosis Category"
   ,category.code as "Legalaid Category Code"
   ,category.name as "Legalaid Category Name"
-  ,category.description as "Legalaid Category Description"
```
Comment (said-moj) [inline cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql]: This one is on the ticket, any reason it was removed?

Comment (TawneeOwl) [reply cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql]: The field returns a description of the legal category selected, the ticket states Operator’s most recent categorisation of client’s legal issue; this is covered in Legalaid Category Name, the full description is not required

File: cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql (hunk 3 above)
Comment (BenMillar-CGI) [reply cla_backend/apps/reports/sql/MIExtractByOutcomeTellUsAboutYourProblem.sql:12]: Thanks @said-moj, I have re-added the source field with commit [6c4dc78](https://github.com/ministryofjustice/cla_backend/pull/1143/commits/6c4dc78008ba44a16cfc300cfd6bc051d7689154).

**Review by digitalali-moj — APPROVED** [review]
Nicely done, thanks  👍

**Review by BenMillar-CGI — APPROVED** [review]

**Pull request merged**